In [1]:
import requests
import pandas as pd
import seaborn as sns

**1.** Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches?  

In [2]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json'
params = {
    'year': '2016',
    'query_text': 'fire'
}
response = requests.get(endpoint, params = params)
response

<Response [200]>

In [3]:
fire = response.json()
fire

[{'month_name': 'January',
  'year': '2016',
  'query_count': '19',
  'query_text': 'fire'},
 {'month_name': 'February',
  'year': '2016',
  'query_count': '35',
  'query_text': 'fire'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '26',
  'query_text': 'fire'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'June',
  'year': '2016',
  'query_count': '31',
  'query_text': 'fire'},
 {'month_name': 'July',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '47',
  'query_text': 'fire'},
 {'month_name': 'September',
  'year': '2016',
  'query_count': '36',
  'query_text': 'fire'},
 {'month_name': 'October',
  'year': '2016',
  'query_count': '38',
  'query_text': 'fire'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '32',
  'query_text

In [4]:
fire_searches = pd.DataFrame({'Month': [fire[x].get('month_name') for x in range(0, 12)], 'Searches': [fire[x].get('query_count') for x in range(0, 12)]})
fire_searches['Month'] = fire_searches['Month'].astype('string')
fire_searches['Searches'] = fire_searches['Searches'].astype('int')
fire_searches


Month  Searches
0     January        19
1    February        35
2       March        32
3       April        26
4         May        24
5        June        31
6        July        24
7      August        47
8   September        36
9     October        38
10   November        32
11   December        38

In [5]:
total = fire_searches['Searches'].sum()
total

382

In [6]:
fire_searches[fire_searches['Searches']==fire_searches['Searches'].max()]

Month  Searches
7  August        47

**A1:** There were 382 searches for "fire" in 2016, with August being the highest, at 47.

**Q2.** Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur?  

In [7]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=50000&&$where=query_count>100.0'
#Socrata limits queries to 1,000 rows by default - maximum limit supported is 50,000, which is enough for this entire data set.
response = requests.get(endpoint)
response

<Response [200]>

In [8]:
res = response.json()
len(res)

1262

**A2.** There are 1,262 occurances of a query being run more than 100 times in a month.

**Q3.** Make another API request that returns all the times "codes" was searched more than 100 times in a month. How many times did this occur? 

In [9]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=50000&$where=query_count>100.0&query_text=codes'
response = requests.get(endpoint)
response

<Response [200]>

In [10]:
codes = response.json()
len(codes)

56

**A3.** There are 56 occurances of "codes" being queried more than 100 times in a month.

**Q4.** Make an API request that returns the entire Top 500 Monthly Searches dataset. Make a chart that shows the number of times "maps" was searched in a month across the entire time frame.

In [11]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=50000&query_text=Maps'
response = requests.get(endpoint)
response

<Response [200]>

In [12]:
maps = response.json()
maps_searches = pd.DataFrame(maps)
maps_searches

month_name  year query_count query_text month
0     January  2014          12       Maps   NaN
1       April  2014          11       Maps   NaN
2         May  2014          10       Maps   NaN
3        July  2014          12       Maps   NaN
4      August  2014           8       Maps   NaN
5   September  2014           7       Maps   NaN
6     October  2014           8       Maps   NaN
7    November  2014           9       Maps   NaN
8     January  2015          15       Maps   NaN
9       April  2015          18       Maps   NaN
10   February  2015          13       Maps   NaN
11        May  2015          12       Maps   NaN
12    January  2017          17       Maps   NaN
13       July  2015          12       Maps   NaN
14       June  2015          20       Maps   NaN
15     August  2015          12       Maps   NaN
16  September  2015           9       Maps   NaN
17    October  2015           8       Maps   NaN
18   November  2015          11       Maps   NaN
19    January  2016          10       Maps   NaN
20   February  2016          13       Maps   NaN
21      March  2016          13       Maps   NaN
22      April  2016          13       Maps   NaN
23        May  2016          20       Maps   NaN
24       June  2016          13       Maps   NaN
25       July  2016          14       Maps   NaN
26     August  2016          15       Maps   NaN
27  September  2016           8       Maps   NaN
28    October  2016          15       Maps   NaN
29   November  2016          19       Maps   NaN
30   December  2016          18       Maps   NaN
31   February  2017          22       Maps   NaN
32      March  2017          18       Maps   NaN
33      April  2017          38       Maps   NaN
34        May  2017          24       Maps   NaN
35       June  2017          26       Maps   NaN
36       July  2017          32       Maps   NaN
37     August  2017          36       Maps   NaN
38  September  2017          25       Maps   NaN
39    October  2017          23       Maps   NaN
40   November  2017          16       Maps   NaN
41   December  2017          29       Maps   NaN
42    January  2018          23       Maps   NaN
43      March  2018          38       Maps   NaN
44      April  2018          34       Maps   NaN
45        May  2018          54       Maps   NaN
46       June  2018          35       Maps   NaN
47       July  2018          34       Maps   NaN
48     August  2018          25       Maps   NaN
49  September  2018          37       Maps   NaN
50    October  2018          18       Maps   NaN
51   November  2018          32       Maps   NaN
52   December  2019         126       Maps    12
53  September  2020         134       Maps     9
54    January  2021         121       Maps     1
55   February  2021         121       Maps     2
56        May  2021         129       Maps     5